evm_arithmetization/src/prover.rs

```diff
@@ -30,132 +30,148 @@ use starky::stark::Stark;
 
 use crate::all_stark::{AllStark, Table, NUM_TABLES};
 use crate::cpu::kernel::aggregator::KERNEL;
-use crate::cpu::kernel::interpreter::{generate_segment, ExtraSegmentData};
+use crate::cpu::kernel::interpreter::{
+    generate_segment, generate_segment_given_data, ExtraSegmentData,
+};
 use crate::generation::state::GenerationState;
 use crate::generation::{generate_traces, GenerationInputs, MemBeforeValues, SegmentData};
 use crate::get_challenges::observe_public_values;
 use crate::memory::segments::Segment;
 use crate::proof::{AllProof, MemCap, PublicValues, RegistersData};
-use crate::witness::memory::MemoryAddress;
+use crate::witness::memory::{MemoryAddress, MemoryState};
 use crate::witness::state::RegistersState;
 
+/// Structure holding the data needed to initialize a segment.
+#[derive(Clone, Default, Debug)]
+pub struct GenerationSegmentData {
+    /// Registers at the start of the segment execution.
+    pub registers: RegistersState,
+    /// Memory at the start of the segment execution.
+    pub(crate) memory: MemoryState,
+    /// Extra data required to initialize a segment.
+    pub(crate) extra_data: ExtraSegmentData,
+}
+
+/// Alias for `RegistersState`.
+pub type Registers = RegistersState;
```

Comment:
Given that this is used only at a single place (`generate_txn_proof`), and that the type alias doesn't bring much here anyway, I'd suggest removing, unless you had something specific in mind?

Reply:
Well, the reason why I introduced it was because the decoder doesn't have access to state.rs, and I wasn't sure it would have made sense ro add it on the decoder side... But I can do that instead if you prefer